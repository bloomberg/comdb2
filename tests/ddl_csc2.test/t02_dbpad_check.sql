create table t {
    schema {
        byte a[2]
    }
};$$

# not allowed: need dbpad to change size
alter table t {
    schema {
        byte a[4]
    }
};$$

# not allowed: need dbpad to change size
alter table t {
    schema {
        byte a[1]
    }
};$$

# not allowed: need dbpad to change size
alter table t {
    schema {
        byte a[4] dbstore=0
    }
};$$

# should pass
alter table t {
    schema {
        byte a[3] dbpad=0
    }
};$$

create table t2 {
    schema {
        byte a[2]
    }
};$$


# should pass
alter table t2 {
    schema {
        byte a[2]
        int b dbstore=0
    }
};$$

drop table t;
drop table t2;
