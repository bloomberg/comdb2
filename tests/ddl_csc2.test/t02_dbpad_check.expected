[alter table t {
    schema {
        byte a[4]
    }
}] failed with rc 240 cannot change size of byte array without dbpad
[alter table t {
    schema {
        byte a[1]
    }
}] failed with rc 240 cannot change size of byte array without dbpad
[alter table t {
    schema {
        byte a[4] dbstore=0
    }
}] failed with rc 240 cannot change size of byte array without dbpad
